### PR TITLE
fix: add ledger transactions for bulk usage events

### PR DESCRIPTION
**Pull request by BrooksFlannery**
---
## Summary by cubic
Bulk bulk-inserted usage events now create ledger transactions per new event, fixing missing accounting entries and aligning with the single-event flow. Tests cover idempotency and ensure commands only generate for newly inserted events.

- **Bug Fixes**
  - Generate and persist UsageEventProcessed ledger transactions during bulk insert (idempotent by transactionId).
  - Use the comprehensive authenticated transaction wrapper to execute ledger commands safely. Default missing properties and usageDate.
  - Add tests for slug resolution, idempotency, and correct linkage to usage events and subscriptions.

<sup>Written for commit e6cfb58d67cf8aef42cd72d168c94e9326e7ce40. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **New Features**
  * Bulk insert now returns generated ledger commands along with created usage events; event properties and dates default when omitted.

* **Bug Fixes**
  * Improved validation and error handling to prevent invalid or duplicate events and ensure ledger entries are recorded reliably, including error when no identifier is provided.

* **Tests**
  * Added comprehensive transaction-level tests for bulk insert (slug/id resolution, idempotency, multi-customer, negative paths); simplified some existing test scenarios.

* **Refactor**
  * Bulk-insert flow centralized for clearer, more predictable behavior.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: platform/flowglad-next/src/server/routers/usageEventsRouter.test.ts [hunk 1]
```diff
@@ -1121,6 +1122,52 @@ describe('usageEventsRouter', () => {
       expect(result.usageEvents[1].amount).toBe(200)
       expect(result.usageEvents[2].amount).toBe(150)
       expect(result.usageEvents[3].amount).toBe(250)
+
+      // Verify ledger transactions were created for all bulk inserted events
+      await authenticatedTransaction(
```
Comment (agreea) [inline platform/flowglad-next/src/server/routers/usageEventsRouter.test.ts]: Do we want this to be tested in the router tests or should we be testing this in the more pure business logic tests, to avoid all the performance overhead / timeout issues we see with the router tests?

Also do we want to test that it's deduplicating properly and not inserting multiple ledger transactions for effectively duplicate events? Or is that going to be something we test in a subsequent PR?

File: platform/flowglad-next/src/utils/usage/bulkInsertUsageEventsTransaction.test.ts [hunk 2]
```diff
@@ -0,0 +1,708 @@
...
+          },
+          transaction
+        )
+      )
+
+      expect(result.result.usageEvents).toHaveLength(1)
+      expect(result.result.usageEvents[0].properties).toEqual({})
+    })
+
+    it('should default usageDate to current timestamp', async () => {
```
[the remark this answers is not in the document]
Comment (agreea) [reply platform/flowglad-next/src/utils/usage/bulkInsertUsageEventsTransaction.test.ts]: +1

File: platform/flowglad-next/src/utils/usage/bulkInsertUsageEventsTransaction.test.ts [hunk 3]
```diff
@@ -0,0 +1,708 @@
...
+                ],
+              },
+              livemode: true,
+            },
+            transaction
+          )
+      )
+
+      expect(firstResult.result.usageEvents).toHaveLength(2)
+      expect(firstResult.ledgerCommands?.length).toBeGreaterThan(0)
```
[the remark this answers is not in the document]
Comment (agreea) [reply platform/flowglad-next/src/utils/usage/bulkInsertUsageEventsTransaction.test.ts]: +1

File: platform/flowglad-next/src/utils/usage/bulkInsertUsageEventsTransaction.test.ts [hunk 4]
```diff
@@ -0,0 +1,708 @@
...
+
+      // Should only insert the new event
+      expect(secondResult.result.usageEvents).toHaveLength(1)
+      expect(secondResult.result.usageEvents[0].amount).toBe(300)
+      // Should only generate ledger commands for the new event (not for the duplicate)
+      expect(secondResult.ledgerCommands?.length).toBeGreaterThan(0)
+      // The first result should have generated commands for 2 events
+      expect(firstResult.ledgerCommands?.length).toBeGreaterThan(0)
+      // The second result should only have commands for 1 new event (the duplicate should not generate commands)
+      // This verifies that deduped entries don't generate ledger commands
```
[the remark this answers is not in the document]
Comment (agreea) [reply platform/flowglad-next/src/utils/usage/bulkInsertUsageEventsTransaction.test.ts]: +1

File: platform/flowglad-next/src/utils/usage/bulkInsertUsageEventsTransaction.test.ts [hunk 5]
```diff
@@ -0,0 +1,708 @@
...
+          },
+          transaction
+        )
+      )
+
+      expect(result.result.usageEvents).toHaveLength(2)
+      expect(result.result.usageEvents[0].amount).toBe(100)
+      expect(result.result.usageEvents[1].amount).toBe(200)
+      expect(result.ledgerCommands?.length).toBeGreaterThan(0)
+    })
```
[the remark this answers is not in the document]
Comment (agreea) [reply platform/flowglad-next/src/utils/usage/bulkInsertUsageEventsTransaction.test.ts]: 1+

File: platform/flowglad-next/src/utils/usage/bulkInsertUsageEventsTransaction.test.ts [hunk 6]
```diff
@@ -0,0 +1,708 @@
...
+      result.result.usageEvents.forEach((event) => {
+        expect(event.priceId).toBe(price.id)
+      })
+
+      // Verify amounts match the input
+      expect(result.result.usageEvents[0].amount).toBe(100)
+      expect(result.result.usageEvents[1].amount).toBe(200)
+      expect(result.result.usageEvents[2].amount).toBe(150)
+      expect(result.result.usageEvents[3].amount).toBe(250)
+    })
```
[the remark this answers is not in the document]
Comment (agreea) [reply platform/flowglad-next/src/utils/usage/bulkInsertUsageEventsTransaction.test.ts]: +1
